python/cudf_polars/cudf_polars/dsl/ir.py

```diff
@@ -846,9 +846,10 @@ class MapFunction(IR):
 
     _NAMES: ClassVar[frozenset[str]] = frozenset(
         [
-            "drop_nulls",
             "rechunk",
-            "merge_sorted",
+            # libcudf merge is not stable wrt order of inputs, since
+            # it uses a priority queue to manage the tables it produces.
+            # "merge_sorted",
```

Comment:
Can you explain this a bit more, and perhaps file an issue for the behavior you want?

Would you achieve the stability you seek by adding a column containing the table index and appending that to the end of the key columns on which this is comparing while doing the merge? i.e. the LHS table gets a column of zeroes and the RHS gets a column of ones, and then we sort based on `key_columns=[*whatever_keys_the_user_gave, table_index]`? If that's an accurate summary, I think we may have enough information to do that in libcudf without allocating any new columns or incurring much of a performance hit.

Reply:
I'll open an issue.

Reply:
https://github.com/rapidsai/cudf/issues/16010

Unfortunately I don't think my proposal works for the n-ary case :(. But polars merge is only binary, so it's ok.

Reply:
The implementation I had in mind should be safe/correct for n-ary cases? I will check to make sure.

Reply:
Yes, it would be. I was hoping to be able to avoid the extra column (even if it is only in the output intermediates)